### PR TITLE
All 5 pipelines now passing preliminary tests on AWS and populating RDS with real data & StarWatch dashboard running as a service with basic Stellarium integration. 

### LINKED ISSUE #16: Create weekly meteo API extract script
As a developer, when dealing with the meteo API, I want to extract relevant specific information about sunrise schedules to a local format using API requests so that the data can then be manipulated and processed using python.

This should extract information about:
- [ ] Sunrise data

### PR DESCRIPTION
**Pull request by MurasatoNaoya**
### This pull request is a collection of changes that irons out all of the bugs that came with deploying all pipelines to AWS and running the dashboard as a persistent ECS service. 

-  A sole pipeline file for the weekly bodies pipeline, in a similar format to the AuroraWatch pipeline. Building on @olliejth's work.
-  A Dockerfile for the deployment of the weekly bodies pipeline as a container to its associated ECR.
-  An amended Terraform file that accounts for the IAM role of CloudWatch that enables Step Function Logging.
-  A base dashboard implementation, that includes a crude Stellarium integration.
- A Dockerfile for the deployment of the dashboard to run as a service (still running as a service as of writing, available at http://13.42.55.67:8501/ until further notice). 

This closes #16 #176 #63 #182 #54 #47 #186 #187 #188

### CODE REVIEW

File: astronomy-pipeline/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+COPY requirements.txt .
+RUN pip install -r requirements.txt
+
+COPY astronomy_load.py .
+COPY astronomy_extract.py .
+COPY astronomy_extract_functions.py .
+COPY astronomy_transform.py .
+COPY astronomy_transform_functions.py .
+COPY astronomy_pipeline.py .
```
Comment (AlexOBarnes) [inline astronomy-pipeline/Dockerfile:14]: This should be changed in a future pull request

File: dashboard/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+COPY . /app
+
+# Install any needed packages specified in requirements.txt
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Expose the default port that Streamlit uses
+EXPOSE 8501
+
+# Command to run your app
+CMD ["streamlit", "run", "dashboard.py", "--server.port=8501", "--server.enableCORS=false", "--server.enableXsrfProtection=false"]
```
Comment (AlexOBarnes) [inline dashboard/Dockerfile:17]: Really interesting CMD, make sure it is documented

File: dashboard/dashboard.py [hunk 3]
```diff
@@ -1 +1,106 @@
...
+st.title("⭐ Starwatch Data Dashboard ⭐")
+
+
+st.markdown("""
+This dashboard provides an interactive view of the night sky as seen from **Liverpool Street, London**. The view updates automatically to reflect the current date and time.
+""")
+
+# Default Coordinates for Liverpool Street, London from Google Maps.
+DEFAULT_LATITUDE = 51.5171
+DEFAULT_LONGITUDE = -0.0822
```
Comment (AlexOBarnes) [inline dashboard/dashboard.py:16]: Security issue?

File: terraform/variables.tf [hunk 4]
```diff
@@ -91,7 +91,7 @@ variable "NASA_API_KEY" {
 
 
 
-variable "ASTRONOMY_KEY_ID" {
+variable "ASTRONOMY_ID" {
```
Comment (AlexOBarnes) [inline terraform/variables.tf:94]: Very informative naming convention

File: dashboard/dashboard.py [hunk 5]
```diff
@@ -1 +1,106 @@
-'''Dashboard for starwatch data'''
+import streamlit as st
+from datetime import datetime
+import urllib.parse
+from streamlit_autorefresh import st_autorefresh
+
+
+st.title("⭐ Starwatch Data Dashboard ⭐")
```
Comment (AlexOBarnes) [inline dashboard/dashboard.py:7]: Love the emojis

**Review by AlexOBarnes — APPROVED** [review]
Very good

File: astronomy-pipeline/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,19 @@
+
```
Comment (CalebMathers) [inline astronomy-pipeline/Dockerfile:1]: Does this empty line need to be here?

File: astronomy-pipeline/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+COPY astronomy_load.py .
+COPY astronomy_extract.py .
+COPY astronomy_extract_functions.py .
+COPY astronomy_transform.py .
+COPY astronomy_transform_functions.py .
+COPY astronomy_pipeline.py .
+
+
+
```
Comment (CalebMathers) [inline astronomy-pipeline/Dockerfile:17]: Lots of whitespace

File: astronomy-pipeline/astronomy_load.py [hunk 8]
```diff
@@ -52,4 +52,4 @@ def upload_astronomy_data(data_dict: list[dict]) -> None:
     moon_data = data_dict["moon_phase_list"]
 
     upload_body_position_data(position_data)
-    upload_moon_phase_data(moon_data)
+    upload_moon_phase_data(moon_data)
```
Comment (CalebMathers) [inline astronomy-pipeline/astronomy_load.py:55]: Good change?

File: astronomy-pipeline/astronomy_pipeline.py [hunk 9]
```diff
@@ -1,11 +1,15 @@
...
 
 from astronomy_extract import extract_weekly_astronomy_data
 from astronomy_transform import transform_astronomy_data
 from astronomy_load import upload_astronomy_data
 
-if __name__ == "__main__":
 
+def lambda_handler(event=None, context=None) -> None:
+    '''Runs the notification pipeline'''
+    print('Running Lambda')
```
Comment (CalebMathers) [inline astronomy-pipeline/astronomy_pipeline.py:12]: Could this print statement be removed in favour of logging now it's going to prod?

File: astronomy-pipeline/Dockerfile (hunk 1 above)
Comment (olliejth) [reply astronomy-pipeline/Dockerfile:14]: New change has been discussed and should be solved by submitted request.

File: dashboard/Dockerfile (hunk 2 above)
Comment (CalebMathers) [reply dashboard/Dockerfile:17]: True true

File: astronomy-pipeline/astronomy_transform.py [hunk 10]
```diff
@@ -1,5 +1,8 @@
-"""Transform script to reshape Astronomy API data into a format
-for upload to the RDS database."""
+"""
+astronomy_transform
+Transform script to reshape Astronomy API data into a format
```
Comment (olliejth) [inline astronomy-pipeline/astronomy_transform.py:3]: Nice addition

File: astronomy-pipeline/astronomy_transform_functions.py [hunk 11]
```diff
@@ -1,4 +1,4 @@
-"""Functions for the astronomy API transform script."""
+"""astronomy_transform_functions, functions for the astronomy API transform script."""
```
Comment (olliejth) [inline astronomy-pipeline/astronomy_transform_functions.py:2]: Nice standardisation

File: dashboard/dashboard.py (hunk 5 above)
Comment (CalebMathers) [reply dashboard/dashboard.py:7]: Beautiful

File: astronomy-pipeline/astronomy_transform_functions.py [hunk 12]
```diff
@@ -80,23 +80,20 @@ def clean_position_data(df: pd.DataFrame) -> list:
     to more appropriate data types."""
 
     # Get rid of earth entries
-    df = df[df["body_name"] != "earth"]
+    df = df[df["body_name"] != "earth"].copy()  # Create a copy to avoid warnings
 
     # Gets body_IDs from database
     body_mapping = get_body_mapping()
-    df["body_id"] = df["body_name"].map(body_mapping)
+    df.loc[:, "body_id"] = df["body_name"].map(body_mapping)
```
Comment (olliejth) [inline astronomy-pipeline/astronomy_transform_functions.py:88]: Thank you for fixing this.

File: dashboard/dashboard.py (hunk 3 above)
Comment (CalebMathers) [reply dashboard/dashboard.py:16]: Lets hope Russia doesn't see this

File: dashboard/Dockerfile (hunk 2 above)
Comment (olliejth) [inline dashboard/Dockerfile:17]: Nice secure way of doing this.

File: dashboard/requirements.txt [hunk 13]
```diff
@@ -1,2 +1,5 @@
 pytest
-pylint
\ No newline at end of file
+pylint
+altair
+streamlit
+streamlit_autorefresh
```
Comment (CalebMathers) [inline dashboard/requirements.txt:5]: Interesting way to refresh the dashboard automatically

File: dashboard/dashboard.py [hunk 14]
```diff
@@ -1 +1,106 @@
...
+        }}
+    </style>
+    <iframe src="{stellarium_url}" scrolling="no"></iframe>
+    <script>
+        document.querySelector('iframe').onload = function() {{
+            const iframeDoc = this.contentDocument || this.contentWindow.document;
+            iframeDoc.querySelector('aside').style.display = 'none';  // Hides the left menu
+        }};
+    </script>
+    """, height=600)
```
Comment (olliejth) [inline dashboard/dashboard.py:97]: Very nice addition.

File: terraform/main.tf
Comment (CalebMathers) [inline terraform/main.tf:1]: Incredible terraform, I'd say it terrafic!

**Review by CalebMathers — APPROVED** [review]

File: astronomy-pipeline/astronomy_load.py (hunk 8 above)
Comment (MurasatoNaoya) [reply astronomy-pipeline/astronomy_load.py:55]: Certified bruh moment.

**Pull request merged**